plugins/woocommerce-blocks/assets/js/base/components/combobox/index.tsx

```diff
@@ -65,6 +100,28 @@ const Combobox = ( {
 		};
 	} );
 
+	const initialOption = options.find( ( option ) => option.value === value );
+
+	const [ searchTerm, setSearchTerm ] = useState(
+		initialOption?.label || ''
+	);
+
+	useEffect( () => {
+		setSearchTerm( initialOption?.label || '' );
+	}, [ initialOption ] );
+
+	const [ selectedOption, setSelectedOption ] = useState( initialOption );
```

Comment:
```suggestion
	const [ selectedOption, setSelectedOption ] = useState( initialOption );

	// If the list of options changes and the currently selected option is no longer available,
	// we should unset the search term and deselect the option.
	useEffect( () => {
		if (
			options.length > 0 &&
			! options.some( ( option ) => option.value === value )
		) {
			setSearchTerm( initialOption?.label || '' );
			setSelectedOption( initialOption );
		}
	}, [ initialOption, value, options ] );
```

Reply:
Ah nice catch! I hadn't considered the situation where the options change. I'll apply this. ty 🙇🏻